### PR TITLE
Tag traces with HTTP headers from DD_TRACE_HEADER_TAGS

### PR DESCRIPTION
**Pull request by nizox**
Other APM tracers use the `DD_TRACE_HEADER_TAGS` environment variable to tag traces with HTTP header values (e.g. [Java documentation](https://docs.datadoghq.com/tracing/setup_overview/setup/java/?tab=containers#configuration)). This PR implements the same mechanism in a complementary way to the header tracing configured from the code ([documented here](https://ddtrace.readthedocs.io/en/stable/advanced_usage.html#headers-tracing)).

The environment variable must be either comma or space separated. e.g. `Host:http.host,User-Agent:http.user_agent`
or `referer:http.referer Content-Type:http.content_type`. Header names are case insensitive.

## Checklist
- [ ] Added to the correct milestone.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: ddtrace/settings/config.py [hunk 1]
```diff
@@ -210,6 +210,8 @@ def header_tag_name(self, header_name):
         # type: (str) -> Optional[str]
         """
         Returns the tag associated with the current header if it should be traced.
+        An empty string is returned when the current header must be retained but
+        no tag was configured.
```
Comment (P403n1x87) [inline ddtrace/settings/config.py:214]: Is it possible to somehow reduce these duplicate docstrings by abstracting (e.g. a "header" mixin)?

Comment (nizox) [reply ddtrace/settings/config.py:214]: Probably yes but this method is temporary and should be removed with https://github.com/DataDog/dd-trace-py/pull/2762.

**Review by P403n1x87 — DISMISSED** [review]

File: ddtrace/settings/http.py [hunk 2]
```diff
@@ -44,30 +46,36 @@ def trace_headers(self, whitelist):
...
+            self.header_tags.setdefault(normalized_header_name, "")
 
         # Mypy can't catch cached method's invalidate()
-        self.header_is_traced.invalidate()  # type: ignore[attr-defined]
+        self.header_tag_name.invalidate()  # type: ignore[attr-defined]
 
         return self
 
     @cachedmethod()
-    def header_is_traced(self, header_name):
```
Comment (brettlangdon) [inline ddtrace/settings/http.py:55]: are we able to get rid of this API?

technically it is public

File: ddtrace/settings/config.py [hunk 3]
```diff
@@ -202,14 +206,17 @@ def trace_headers(self, whitelist):
         self.http.trace_headers(whitelist)
         return self
 
-    def header_is_traced(self, header_name):
```
Comment (brettlangdon) [inline ddtrace/settings/config.py:205]: this API is public, I don't think we can get rid of it

File: ddtrace/settings/integration.py [hunk 4]
```diff
@@ -86,18 +86,20 @@ def is_header_tracing_configured(self):
         """
         return self.http.is_header_tracing_configured or self.global_config.http.is_header_tracing_configured
 
-    def header_is_traced(self, header_name):
```
Comment (brettlangdon) [inline ddtrace/settings/integration.py:89]: same here

**Review by brettlangdon — CHANGES_REQUESTED** [review]

File: ddtrace/settings/http.py (hunk 2 above)
Comment (nizox) [reply ddtrace/settings/http.py:55]: I added back the public method so we do not break the compatibility.

File: ddtrace/settings/config.py (hunk 3 above)
Comment (nizox) [reply ddtrace/settings/config.py:205]: I added back the public method so we do not break the compatibility.

File: ddtrace/settings/integration.py (hunk 4 above)
Comment (nizox) [reply ddtrace/settings/integration.py:89]: I added back the public method so we do not break the compatibility.

File: ddtrace/profiling/collector/stack.pyx [hunk 5]
```diff
@@ -21,6 +21,7 @@ from ddtrace.profiling.collector import _traceback
 from ddtrace.utils import attr as attr_utils
 from ddtrace.utils import formats
 
+
```
Comment (brettlangdon) [inline ddtrace/profiling/collector/stack.pyx:24]: unnecessary change?

**Review by brettlangdon — APPROVED** [review]
minor comment, but otherwise lgtm

**Review by Kyle-Verhoog — APPROVED** [review]

**Comment by brettlangdon** [comment]
@Mergifyio backport 0.53
